src/explorer.api/Exploration/ExplorationRegistry.cs

```diff
@@ -20,26 +27,19 @@ public Guid Register(Exploration exploration, CancellationTokenSource tokenSourc
             return id;
         }
 
-        public async Task Remove(Guid id)
+        public void Remove(Guid id)
         {
-            registrations.Remove(id, out var registration);
-
-            try
+            if (!registrations.TryGetValue(id, out var registration))
             {
-                if (registration is null)
-                {
-                    return;
-                }
-                registration.CancelExploration();
-
-                // Wait for the task to finish before disposing the registration.
-                // This will also allow any Exceptions to surface.
-                await registration.Exploration.Completion;
+                return;
             }
-            finally
+
+            if (registration.Status == ExplorationStatus.Processing)
             {
-                registration?.Dispose();
+                registration.CancelExploration();
```

Comment:
My feeling is that the registry should not cancel the exploration, but instead just throw some `InvalidOperationException` in this case.

Reply:
Ok, that makes sense. It should be an error to remove an exploration while it's still running.

Reply:
Done
c8a0966